jax_distribution/systems/ff_ippo_rware.py

```diff
@@ -317,15 +319,133 @@ def learner_fn(runner_state: RunnerState) -> ExperimentOutput:
             _update_step, in_axes=(0, None), axis_name="batch"
         )
 
-        runner_state, metric = jax.lax.scan(
-            batched_update_step, runner_state, None, config["NUM_UPDATES"]
+        runner_state, (metric, loss_info) = jax.lax.scan(
+            batched_update_step, runner_state, None, config["NUM_UPDATES_PER_EVAL"]
         )
-        return {"runner_state": runner_state, "metrics": metric}
+        total_loss, (value_loss, loss_actor, entropy) = loss_info
+        return {
+            "runner_state": runner_state,
+            "metrics": metric,
+            "total_loss": total_loss,
+            "value_loss": value_loss,
+            "loss_actor": loss_actor,
+            "entropy": entropy,
+        }
 
     return learner_fn
 
 
-def learner_setup(env: Environment, config: Dict) -> Tuple[callable, RunnerState]:
+def get_evaluator_fn(env: Environment, apply_fn: callable, config: dict) -> callable:
+    """Get the evaluator function."""
+
+    def eval_one_episode(params, runner_state) -> Tuple:
+        """Evaluate one episode. It is vectorized over the number of evaluation episodes."""
+
+        def _env_step(runner_state: Tuple) -> Tuple:
```

Comment:
Seems like this is duplicate code of [here](https://github.com/instadeepai/Mava/blob/1e9f67a5c9a1756a59f51598fc8b2bcd60729135/jax_distribution/systems/ff_ippo_rware.py#L122), if you pulled out to a common method you'd just need to pass in params, runner state and eval mode

Also you have a runner state type, so should that not be used instead of tuple?